specification/communication/Communication.JobRouter/models.tsp

```diff
@@ -0,0 +1,1198 @@
+  value?: unknown;
+
+  @doc("The type discriminator describing a sub-type of Rule")
+  kind: "static-rule";
+}
+
+@doc("Describes a worker selector that will be attached to the job")
+model StaticWorkerSelectorAttachment extends WorkerSelectorAttachment {
+  @doc("The worker selector to attach.")
+  workerSelector: RouterWorkerSelector;
+
+  @doc("The type discriminator describing the type of worker selector attachment")
+  kind: "static";
+}
+
+@doc("Trigger for an exception action on exceeding wait time")
+model WaitTimeExceptionTrigger extends ExceptionTrigger {
+  @doc("Threshold for wait time for this trigger.")
+  thresholdSeconds: float64;
+
+  @doc("The type discriminator describing a sub-type of ExceptionTrigger")
+  kind: "wait-time";
+}
+
+@doc("A rule providing a binding to an external web server.")
+model WebhookRouterRule extends RouterRule {
+  @doc("Uri for Authorization Server.")
+  authorizationServerUri?: url;
+
+  @doc("OAuth2.0 Credentials used to Contoso's Authorization server. Reference: https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/")
+  clientCredential?: OAuth2WebhookClientCredential;
+
+  @doc("Uri for Contoso's Web Server.")
+  webhookUri?: url;
+
+  @doc("The type discriminator describing a sub-type of Rule")
+  kind: "webhook-rule";
+}
+
+@doc("Describes multiple sets of queue selectors, of which one will be selected and attached according to a weighting")
+model WeightedAllocationQueueSelectorAttachment
+  extends QueueSelectorAttachment {
+  @doc("A collection of percentage based weighted allocations.")
+  allocations: QueueWeightedAllocation[];
+
+  @doc("The type discriminator describing the type of queue selector attachment")
+  kind: "weighted-allocation-queue-selector";
+}
+
+@doc("Describes multiple sets of worker selectors, of which one will be selected and attached according to a weighting")
+model WeightedAllocationWorkerSelectorAttachment
+  extends WorkerSelectorAttachment {
+  @doc("A collection of percentage based weighted allocations.")
+  allocations: WorkerWeightedAllocation[];
+
+  @doc("The type discriminator describing the type of worker selector attachment")
+  kind: "weighted-allocation-worker-selector";
+}
+
+@doc("Contains the weight percentage and worker selectors to be applied if selected for weighted distributions.")
+model WorkerWeightedAllocation {
+  @doc("The percentage of this weight, expressed as a fraction of 1.")
+  weight: float64;
+
+  @doc("A collection of worker selectors that will be applied if this allocation is selected.")
+  workerSelectors: RouterWorkerSelector[];
+}
+
+@access(Access.internal)
+@doc("Request payload for reclassifying jobs")
+model ReclassifyJobOptions is Record<string>;
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "https://github.com/microsoft/typespec/issues/2542"
+@access(Access.internal)
+@doc("Response payload from reclassifying a job")
+model ReclassifyJobResult is Record<unknown>;
```

Comment:
I don't think you want `Record<unknown>` here, as it creates a schema with `additionalProperties: {}`. I think an empty model is all that is needed.
```suggestion
@access(Access.internal)
@doc("Response payload from reclassifying a job")
model ReclassifyJobResult {}
```